scripts/upgrade.sh.in

```diff
@@ -149,4 +160,5 @@ run "Performing Pre-Upgrade checks" precheck_upgrade
 [ $DRY_RUN -eq 1 ] && exit 0
 run "Upgrading bootstrap" upgrade_bootstrap
 run "Setting cluster version to $DESTINATION_VERSION" patch_kubesystem_namespace
+run "Launching the pre-upgrade" launch_pre_upgrade
```

Comment:
Can't we do all this stuff in some orchestration rather than Bash?

Reply:
For upgrade we could but for downgrade we can't, because for downgrade we want to launch this "pre-downgrade" states before downgrading the bootstrap, 
Just prefer to have the same approach for upgrade and downgrade